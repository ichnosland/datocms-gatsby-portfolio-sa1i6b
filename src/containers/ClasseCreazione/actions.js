/*
 *
 * ClasseCreazione actions
 *
 */

import {
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
  CLASSE_CREAZIONE_GEO_SET,
  CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  CLASSE_CREAZIONE_GEO_COMUNI_SET,
  CLASSE_CREAZIONE_DATA_SET,
  CLASSE_CREAZIONE_DATA_POST,
  CLASSE_CREAZIONE_RESET,
  CLASSE_CREAZIONE_SPINNER_SET,
  CLASSE_CREAZIONE_FEEDBACK_SET,
  CLASSE_CREAZIONE_FEEDBACK_RESET,
  CLASSE_CREAZIONE_GEO_SCUOLE_SET,
} from './constants';


export const classeCreazioneScuoleAttiveFetch = () => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
});

export const classeCreazioneScuoleAttiveSet = (payload) => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  payload,
});

export const classeCreazioneGeoProvinciaFetch = () => ({
  type: CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
});

export const classeCreazioneGeoSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SET,
  payload,
});

export const classeCreazioneGeoComuneFetch = (sigla) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  sigla,
});

export const classeCreazioneGeoComuneSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_SET,
  payload,
});

export const classeCreazioneGeoScuoleSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SCUOLE_SET,
  payload,
});

export const classeCreazioneDataSet = (payload) => ({
  type: CLASSE_CREAZIONE_DATA_SET,
  payload,
});

export const classeCreazioneDataPost = (payload, history) => ({
  type: CLASSE_CREAZIONE_DATA_POST,
  payload,
  history,
});

export const classeCreazioneReset = () => ({
  type: CLASSE_CREAZIONE_RESET,
});

export const classeCreazioneSpinnerSet = (enable) => ({
  type: CLASSE_CREAZIONE_SPINNER_SET,
  enable,
});

export const classeCreazioneFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSE_CREAZIONE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classeCreazioneFeedbackReset = () => ({
  type: CLASSE_CREAZIONE_FEEDBACK_RESET,
});
