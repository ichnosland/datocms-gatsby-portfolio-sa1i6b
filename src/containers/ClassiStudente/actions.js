/*
 *
 * ClassiStudente actions
 *
 */

import {
  CLASSI_STUDENTE_FETCH,
  CLASSI_STUDENTE_SET,
  CLASSI_STUDENTE_RESET,
  CLASSI_STUDENTE_ISCRIVITI_POST,
  CLASSI_STUDENTE_SPINNER_SET,
  CLASSI_STUDENTE_FEEDBACK_SET,
  CLASSI_STUDENTE_FEEDBACK_RESET,
} from './constants';

export const classiStudenteFetch = (disciplinaId) => ({
  type: CLASSI_STUDENTE_FETCH,
  disciplinaId,
});

export const classiStudenteSet = (payload) => ({
  type: CLASSI_STUDENTE_SET,
  payload,
});

export const classiStudenteReset = () => ({
  type: CLASSI_STUDENTE_RESET,
});

export const classiStudenteIscrivitiPost = (codiceCorso, configuration) => ({
  type: CLASSI_STUDENTE_ISCRIVITI_POST,
  codiceCorso,
  configuration,
});

export const classiStudenteSpinnerSet = (enable) => ({
  type: CLASSI_STUDENTE_SPINNER_SET,
  enable,
});

export const classiStudenteFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSI_STUDENTE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classiStudenteFeedbackReset = () => ({
  type: CLASSI_STUDENTE_FEEDBACK_RESET,
});
