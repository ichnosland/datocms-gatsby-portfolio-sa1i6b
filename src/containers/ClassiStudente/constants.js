/*
 *
 * ClassiStudente constants
 *
 */

export const CLASSI_STUDENTE_URI_ISCRIVITI_POST = 'athome/api/classe/studente_add';
export const CLASSI_STUDENTE_URI_FETCH = 'athome/api/classe/studente_list/';

export const CLASSI_STUDENTE_FETCH = 'app/containers/ClassiStudente/CLASSI_STUDENTE_FETCH';
export const CLASSI_STUDENTE_SET = 'app/containers/ClassiStudente/CLASSI_STUDENTE_SET';
export const CLASSI_STUDENTE_RESET = 'app/containers/ClassiStudente/CLASSI_STUDENTE_RESET';
export const CLASSI_STUDENTE_ISCRIVITI_POST = 'app/containers/ClassiStudente/CLASSI_STUDENTE_ISCRIVITI_POST';
export const CLASSI_STUDENTE_SPINNER_SET = 'app/containers/ClassiStudente/CLASSI_STUDENTE_SPINNER_SET';
export const CLASSI_STUDENTE_FEEDBACK_SET = 'app/containers/ClassiStudente/CLASSI_STUDENTE_FEEDBACK_SET';
export const CLASSI_STUDENTE_FEEDBACK_RESET = 'app/containers/ClassiStudente/CLASSI_STUDENTE_FEEDBACK_RESET';
