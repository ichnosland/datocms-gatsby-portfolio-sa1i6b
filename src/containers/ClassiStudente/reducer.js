/*
 *
 * ClassiStudente reducer
 *
 */

import { combineReducers } from 'redux-immutable';
import { fromJS } from 'immutable';

import {
  CLASSI_STUDENTE_SPINNER_SET,
  CLASSI_STUDENTE_FEEDBACK_SET,
  CLASSI_STUDENTE_FEEDBACK_RESET,
  CLASSI_STUDENTE_SET,
  CLASSI_STUDENTE_RESET,
} from './constants';

export const defaultClassiStudenteState = fromJS({
  hasFeedback: false,
  tipologia: '',
  messaggio: '',
});

function classiStudenteSpinnerReducer(state = fromJS(false), action) {
  switch (action.type) {
    default:
      return state;

    case CLASSI_STUDENTE_FEEDBACK_SET:
      return fromJS(false);

    case CLASSI_STUDENTE_SPINNER_SET:
      return fromJS(action.enable);
  }
}

function classiStudenteFeedbackReducer(state = defaultClassiStudenteState, action) {
  switch (action.type) {
    default:
      return state;

    case CLASSI_STUDENTE_FEEDBACK_SET:
      return state.merge({
        hasFeedback: action.enable,
        tipologia: action.tipologia,
        messaggio: action.messaggio,
      });

    case CLASSI_STUDENTE_FEEDBACK_RESET:
      return defaultClassiStudenteState;
  }
}

function classiStudenteListReducer(state = fromJS([]), action) {
  switch (action.type) {
    default:
      return state;

    case CLASSI_STUDENTE_SET:
      return state.merge(action.payload);

    case CLASSI_STUDENTE_RESET:
      return fromJS([]);
  }
}

const classiStudenteReducer = combineReducers({
  spinner: classiStudenteSpinnerReducer,
  feedback: classiStudenteFeedbackReducer,
  list: classiStudenteListReducer,
});

export default classiStudenteReducer;
