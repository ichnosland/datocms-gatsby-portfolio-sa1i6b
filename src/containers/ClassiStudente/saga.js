import { put, takeEvery, call } from 'redux-saga/effects';
import axios from 'axios';

import { API_BASE_PATH } from 'configuration';
import { userEvaluateTrigger } from 'containers/User/actions';
import {
  CLASSI_STUDENTE_FETCH,
  CLASSI_STUDENTE_URI_ISCRIVITI_POST,
  CLASSI_STUDENTE_URI_FETCH,
  CLASSI_STUDENTE_ISCRIVITI_POST,
} from './constants';
import {
  classiStudenteFetch,
  classiStudenteSet,
  classiStudenteSpinnerSet,
  classiStudenteFeedbackSet,
  classiStudenteFeedbackReset,
} from './actions';


/**
 * Fa una chiamata post all'endpoint che effettua l'iscrizione
 * dello studente ad un corso
 * @param {object} data dati da inviare
 * @param {object} data.configuration configuration del progetto
 * @param {number} data.configuration.disciplinaId pk della disciplina
 * @param {string} data.codiceCorso codice del corso da inviare
 */
export function* classiStudenteIscrivitiPost(data) {
  yield put(classiStudenteSpinnerSet(true));
  yield put(classiStudenteFeedbackReset());
  try {
    const response = yield call(
      axios.post,
      `${API_BASE_PATH}${CLASSI_STUDENTE_URI_ISCRIVITI_POST}`, {
        corso: data.codiceCorso,
        disciplina: data.configuration.disciplinaId,
      }
    );

    if (response.status === 200) {
      yield put(classiStudenteFeedbackSet(
        true,
        'okay',
        'La tua iscrizione è avvenuta correttamente'
      ));
      yield put(classiStudenteFetch(data.configuration.disciplinaId));
      yield put(userEvaluateTrigger(data.configuration));
    } else {
      yield put(classiStudenteFeedbackSet(
        true,
        'error',
        'Il codice inserito non è valido'
      ));
    }
  } catch (error) {
    yield put(classiStudenteFeedbackSet(
      true,
      'error',
      'Il codice inserito non è valido'
    ));
  }
  yield put(classiStudenteSpinnerSet(false));
}


/**
 * Esegue il fetch dell'elenco delle classi in cui è
 * iscritto lo studente
 * @param {object} data dati della richiesta
 * @param {number} data.disciplinaId id della disciplina
 */
export function* fetchClassiStudente(data) {
  yield put(classiStudenteSpinnerSet(true));
  yield put(classiStudenteFeedbackReset());
  try {
    const response = yield call(
      axios.get,
      `${API_BASE_PATH}${CLASSI_STUDENTE_URI_FETCH}${data.disciplinaId}`
    );
    yield put(classiStudenteSet(response.data));
  } catch (error) {
    yield put(classiStudenteFeedbackSet(
      true,
      'error',
      'Impossibile iscriversi a questa classe'
    ));
  }
  yield put(classiStudenteSpinnerSet(false));
}

export function* watchClassiStudente() {
  yield takeEvery(CLASSI_STUDENTE_ISCRIVITI_POST, classiStudenteIscrivitiPost);
  yield takeEvery(CLASSI_STUDENTE_FETCH, fetchClassiStudente);
}
