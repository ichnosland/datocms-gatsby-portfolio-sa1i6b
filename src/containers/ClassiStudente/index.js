/**
 *
 * ClassiStudente
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Field, reduxForm, reset } from 'redux-form/immutable';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import Page from 'components/Page';
import TopBar from 'components/TopBar';
import Spinner from 'components/Spinner';
import Container from 'components/Container';
import { H2 } from 'components/Heading';
import { CardForm, FormElement } from 'components/FormElements';
import renderedField from 'components/FormElements/Field';
import AlertBanner from 'components/AlertBanner';
import { Button } from 'components/Button';
import { BrandTxt } from 'components/Text';
import P from 'components/Paragraph';

import reducer from './reducer';
import { watchClassiStudente } from './saga';
import { classiStudenteFetch, classiStudenteIscrivitiPost } from './actions';


export const required = (value) => value ? undefined : 'Campo obbligatorio';
export const validate = (values) => {
  const errors = {};
  const re = new RegExp('[0-9]+$');

  if (!values.get('codice')) {
    errors.codice = 'Campo obbligatorio';
  } else if (!values.get('codice').match(re)) {
    errors.codice = 'Inserire un codice corretto';
  }

  return errors;
};


export class ClassiStudenteView extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  componentDidMount() {
    const { onClassiStudenteFetch, configuration, userAuthentication, history } = this.props;
    if (userAuthentication.codiceDaSbloccare) {
      history.push(configuration.homePage);
    } else {
      onClassiStudenteFetch(configuration.disciplinaId);
    }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { onClassiStudentePost, formIscrizioneData, onResetForm, configuration } = this.props;

    onClassiStudentePost(formIscrizioneData.values.codice, configuration);
    onResetForm();
  };

  render() {
    const { configuration, spinner, feedback, list, formIscrizioneData } = this.props;

    return (
      <Page>
        <TopBar
          backNav={{
            url: configuration.homePage,
            enabled: true,
          }}
          pinned
          title="Classi"
          single
        />
        <Container>
          {spinner ? <Spinner /> : (
            <CardForm onSubmit={this.handleSubmit}>
              <H2>Associa classe</H2>
              <P>Inserisci il codice per associarti ad una classe</P>
              {feedback.hasFeedback && feedback.messaggio && (
                <AlertBanner
                  text={feedback.messaggio}
                  actioncolor={feedback.tipologia}
                >
                  {feedback.messaggio}
                </AlertBanner>)
              }
              <FormElement>
                <Field
                  name="codice"
                  component={renderedField}
                  type="text"
                  label="codice"
                  validate={[required]}
                  align="center"
                />
              </FormElement>
              <FormElement>
                <Button
                  actioncolor="okay"
                  full
                  disabled={(formIscrizioneData || /* istanbul ignore next */ {}).syncErrors}
                >Salva
                </Button>
              </FormElement>
              <P margin="40px auto 20px"><BrandTxt><strong>Classi a cui sei iscritto</strong></BrandTxt></P>
              {list.length ? (
                <div>{list.map((corso) => (<P key={`corso_${corso.pk}`}>{corso.nome}</P>))}</div>
              ) :
                (
                  <AlertBanner
                    actioncolor="help"
                  >
                    Non sei iscritto a classi
                  </AlertBanner>
                )}
            </CardForm>
          )
          }
        </Container>
      </Page>
    );
  }
}

ClassiStudenteView.propTypes = {
  spinner: PropTypes.bool.isRequired,
  feedback: PropTypes.shape({
    hasFeedback: PropTypes.bool.isRequired,
    messaggio: PropTypes.string.isRequired,
    tipologia: PropTypes.string.isRequired,
  }).isRequired,
  list: PropTypes.arrayOf(PropTypes.shape({
    pk: PropTypes.number.isRequired,
    nome: PropTypes.string.isRequired,
  })).isRequired,
  configuration: PropTypes.shape({
    disciplinaId: PropTypes.number.isRequired,
    homePage: PropTypes.string.isRequired,
  }).isRequired,
  onClassiStudenteFetch: PropTypes.func.isRequired,
  onClassiStudentePost: PropTypes.func.isRequired,
  onResetForm: PropTypes.func.isRequired,
  formIscrizioneData: PropTypes.object,
  userAuthentication: PropTypes.shape({
    codiceDaSbloccare: PropTypes.bool.isRequired,
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export const mapStateToProps = (state) => ({
  list: state.get('classiStudente').toJS().list,
  feedback: state.get('classiStudente').toJS().feedback,
  spinner: state.get('classiStudente').toJS().spinner,
  configuration: state.get('configuration').toJS(),
  formIscrizioneData: state.get('form').toJS().iscrizioneClasse,
  userAuthentication: state.get('user').toJS().authentication,
});

function mapDispatchToProps(dispatch) {
  return {
    onClassiStudenteFetch: (disciplinaId) => {
      dispatch(classiStudenteFetch(disciplinaId));
    },
    onClassiStudentePost: (codiceCorso, configuration) => {
      dispatch(classiStudenteIscrivitiPost(codiceCorso, configuration));
    },
    onResetForm: () => {
      dispatch(reset('sbloccaCodice'));
    },
  };
}

const ClassiStudente = reduxForm({
  form: 'iscrizioneClasse',
  validate,
})(ClassiStudenteView);

const withConnect = connect(mapStateToProps, mapDispatchToProps);
const withReducer = injectReducer({ key: 'classiStudente', reducer });
const withSaga = injectSaga({ key: 'classiStudente', saga: watchClassiStudente });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(ClassiStudente);
