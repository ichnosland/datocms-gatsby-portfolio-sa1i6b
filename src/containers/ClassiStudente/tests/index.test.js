
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'configureStore';
import { reset } from 'redux-form/immutable';
import { fromJS } from 'immutable';
import { connect } from 'react-redux';
import { compose } from 'redux';

import injectReducer from 'utils/injectReducer';

import TopBar from 'components/TopBar';
import Spinner from 'components/Spinner';
import Container from 'components/Container';
import AlertBanner from 'components/AlertBanner';
import { CardForm } from 'components/FormElements';
import reducerCurryer from 'common/applications/reducer';
import userReducer from 'containers/User/reducer';
import { defaultClassiStudenteState } from '../reducer';
import { classiStudenteFetch, classiStudenteIscrivitiPost } from '../actions';
import ClassiStudente, {
  ClassiStudenteView,
  required,
  validate,
} from '../index';


const configuration = {
  product: 'product',
  hasPremium: true,
  disciplinaId: 123,
  payPalButtonId: '123456789',
  homePage: '/homepage',
};
const store = configureStore({}, {});
const history = {};
const mockProps = {
  configuration,
  spinner: false,
  feedback: {
    hasFeedback: false,
    tipologia: 'okay',
    messaggio: 'messaggio',
  },
  list: [{
    pk: 1,
    nome: 'corso 1',
  }, {
    pk: 2,
    nome: 'corso 2',
  }],
  onClassiStudenteFetch: () => { },
  onClassiStudentePost: () => { },
  onResetForm: () => { },
  formIscrizioneData: {},
  history: {
    push: () => { },
  },
  userAuthentication: {
    codiceDaSbloccare: false,
  },
};

describe('<ClassiStudenteView />', () => {
  it('visualizza il contenuto nel suo stato iniziale', () => {
    const renderedComponent = shallow(
      <ClassiStudenteView
        store={store}
        history={history}
        {...mockProps}
      />
    );
    expect(renderedComponent.find(TopBar).length).toBe(1);
    expect(renderedComponent.find(Container).length).toBe(1);
    expect(renderedComponent.find(CardForm).length).toBe(1);
    expect(renderedComponent.find(Spinner).length).toBe(0);
    expect(renderedComponent.find(AlertBanner).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('visualizza il contenuto quando spinner = true', () => {
    const props = {
      ...mockProps,
      spinner: true,
    };
    const renderedComponent = shallow(
      <ClassiStudenteView
        store={store}
        history={history}
        {...props}
      />
    );
    expect(renderedComponent.find(TopBar).length).toBe(1);
    expect(renderedComponent.find(Container).length).toBe(1);
    expect(renderedComponent.find(CardForm).length).toBe(0);
    expect(renderedComponent.find(Spinner).length).toBe(1);
    expect(renderedComponent.find(AlertBanner).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('visualizza il contenuto quando ha un feedback', () => {
    const props = {
      ...mockProps,
      feedback: {
        ...mockProps.feedback,
        hasFeedback: true,
      },
    };
    const renderedComponent = shallow(
      <ClassiStudenteView
        store={store}
        history={history}
        {...props}
      />
    );
    expect(renderedComponent.find(TopBar).length).toBe(1);
    expect(renderedComponent.find(Container).length).toBe(1);
    expect(renderedComponent.find(CardForm).length).toBe(1);
    expect(renderedComponent.find(Spinner).length).toBe(0);
    expect(renderedComponent.find(AlertBanner).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('visualizza il contenuto quando non ho corsi', () => {
    const props = {
      ...mockProps,
      list: [],
    };
    const renderedComponent = shallow(
      <ClassiStudenteView
        store={store}
        history={history}
        {...props}
      />
    );
    expect(renderedComponent.find(TopBar).length).toBe(1);
    expect(renderedComponent.find(Container).length).toBe(1);
    expect(renderedComponent.find(CardForm).length).toBe(1);
    expect(renderedComponent.find(Spinner).length).toBe(0);
    expect(renderedComponent.find(AlertBanner).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('handleSubmit deve chiamare onClassiStudentePost e onResetForm', () => {
    const props = {
      ...mockProps,
      onClassiStudentePost: jest.fn(),
      onResetForm: jest.fn(),
      formIscrizioneData: {
        values: {
          codice: 'codice123',
        },
      },
    };
    const renderedComponent = shallow(
      <ClassiStudenteView
        store={store}
        history={history}
        {...props}
      />
    );
    expect(props.onClassiStudentePost).not.toHaveBeenCalled();
    const instance = renderedComponent.instance();
    instance.handleSubmit({ preventDefault: () => { } });
    expect(props.onClassiStudentePost).toHaveBeenCalledWith(
      'codice123', configuration
    );
  });

  it('se codiceDaSbloccare == true devo fare history.push della home', () => {
    const props = {
      ...mockProps,
      history: {
        push: jest.fn(),
      },
      userAuthentication: {
        codiceDaSbloccare: true,
      },
    };
    expect(props.history.push).not.toHaveBeenCalled();

    shallow(
      <ClassiStudenteView
        store={store}
        history={history}
        {...props}
      />
    );
    expect(props.history.push).toHaveBeenCalledWith('/homepage');
  });
});

describe('<ClassiStudente />', () => {
  const withReducerConfiguration = injectReducer({ key: 'configuration', reducer: reducerCurryer(configuration) });
  const withReducerUser = injectReducer({ key: 'user', reducer: userReducer });

  const ConfigurationWrapper = compose(
    withReducerConfiguration,
    withReducerUser,
    connect()
  )(ClassiStudente);

  it('controllo che le props iniziali del componente siano settate correttamente', () => {
    const renderedComponent = shallow(
      <ConfigurationWrapper />, { context: { store } }
    ).dive().dive().dive().dive().dive().dive();
    const receivedProps = renderedComponent.props();

    expect(receivedProps.feedback).toEqual(defaultClassiStudenteState.toJS());
    expect(receivedProps.spinner).toEqual(false);
    expect(receivedProps.list).toEqual([]);
  });

  it('controllo le funzioni di mapDispatchToProps', () => {
    const mockStore = store;
    mockStore.dispatch = jest.fn();
    const renderedComponent = shallow(
      <ConfigurationWrapper />, { context: { store: mockStore } }
    ).dive().dive().dive().dive().dive().dive();
    const receivedProps = renderedComponent.props();

    receivedProps.onClassiStudenteFetch(123);
    expect(mockStore.dispatch).toHaveBeenCalledWith(
      classiStudenteFetch(123)
    );

    receivedProps.onClassiStudentePost('codice123', configuration);
    expect(mockStore.dispatch).toHaveBeenCalledWith(
      classiStudenteIscrivitiPost('codice123', configuration)
    );

    receivedProps.onResetForm();
    expect(mockStore.dispatch).toHaveBeenCalledWith(
      reset('sbloccaCodice')
    );
  });
});

describe('<required />', () => {
  it('se valore non è settato', () => {
    expect(required(undefined)).toBe('Campo obbligatorio');
  });

  it('se valore non è settato', () => {
    expect(required('valore')).toBe(undefined);
  });
});

describe('<validate />', () => {
  it('se valore non è settato', () => {
    expect(validate(fromJS({
      codice: '',
    }))).toEqual({ codice: 'Campo obbligatorio' });
  });

  it('se regex non fa match', () => {
    expect(validate(fromJS({
      codice: 'aaa',
    }))).toEqual({ codice: 'Inserire un codice corretto' });

    expect(validate(fromJS({
      codice: '333aa',
    }))).toEqual({ codice: 'Inserire un codice corretto' });
  });

  it('se valore è corretto', () => {
    expect(validate(fromJS({
      codice: 'aaaa343242',
    }))).toEqual({});
  });
});
