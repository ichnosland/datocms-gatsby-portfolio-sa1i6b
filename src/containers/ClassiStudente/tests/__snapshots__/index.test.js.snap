// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassiStudenteView /> visualizza il contenuto nel suo stato iniziale 1`] = `
<Page>
  <TopBar
    backArrow={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    backNav={
      Object {
        "enabled": true,
        "url": "/homepage",
      }
    }
    closeBtn={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    lesson={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    pinned={true}
    single={true}
    title="Classi"
    topSwitcher={
      Object {
        "items": Array [],
      }
    }
  />
  <Container>
    <FormElements__CardForm
      onSubmit={[Function]}
    >
      <Heading__H2>
        Associa classe
      </Heading__H2>
      <Paragraph
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        Inserisci il codice per associarti ad una classe
      </Paragraph>
      <FormElements__FormElement>
        <Field
          align="center"
          component={[Function]}
          label="codice"
          name="codice"
          type="text"
          validate={
            Array [
              [Function],
            ]
          }
        />
      </FormElements__FormElement>
      <FormElements__FormElement>
        <Button
          actioncolor="okay"
          full={true}
          theme={
            Object {
              "brand": "#00BBEF",
              "fonts": Object {
                "buttons": Object {
                  "fontFamily": "inherit",
                  "fontWeight": "inherit",
                },
              },
              "radius": Object {
                "buttons": "6px",
              },
            }
          }
        >
          Salva
        </Button>
      </FormElements__FormElement>
      <Paragraph
        margin="40px auto 20px"
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        <Text__BrandTxt>
          <strong>
            Classi a cui sei iscritto
          </strong>
        </Text__BrandTxt>
      </Paragraph>
      <div>
        <Paragraph
          key="corso_1"
          theme={
            Object {
              "brand": "#00BBEF",
              "darken": "#1588AA",
            }
          }
        >
          corso 1
        </Paragraph>
        <Paragraph
          key="corso_2"
          theme={
            Object {
              "brand": "#00BBEF",
              "darken": "#1588AA",
            }
          }
        >
          corso 2
        </Paragraph>
      </div>
    </FormElements__CardForm>
  </Container>
</Page>
`;

exports[`<ClassiStudenteView /> visualizza il contenuto quando ha un feedback 1`] = `
<Page>
  <TopBar
    backArrow={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    backNav={
      Object {
        "enabled": true,
        "url": "/homepage",
      }
    }
    closeBtn={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    lesson={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    pinned={true}
    single={true}
    title="Classi"
    topSwitcher={
      Object {
        "items": Array [],
      }
    }
  />
  <Container>
    <FormElements__CardForm
      onSubmit={[Function]}
    >
      <Heading__H2>
        Associa classe
      </Heading__H2>
      <Paragraph
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        Inserisci il codice per associarti ad una classe
      </Paragraph>
      <AlertBanner
        actioncolor="okay"
        text="messaggio"
        theme={
          Object {
            "brand": "#00BBEF",
            "radius": Object {
              "general": "6px",
            },
          }
        }
      >
        messaggio
      </AlertBanner>
      <FormElements__FormElement>
        <Field
          align="center"
          component={[Function]}
          label="codice"
          name="codice"
          type="text"
          validate={
            Array [
              [Function],
            ]
          }
        />
      </FormElements__FormElement>
      <FormElements__FormElement>
        <Button
          actioncolor="okay"
          full={true}
          theme={
            Object {
              "brand": "#00BBEF",
              "fonts": Object {
                "buttons": Object {
                  "fontFamily": "inherit",
                  "fontWeight": "inherit",
                },
              },
              "radius": Object {
                "buttons": "6px",
              },
            }
          }
        >
          Salva
        </Button>
      </FormElements__FormElement>
      <Paragraph
        margin="40px auto 20px"
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        <Text__BrandTxt>
          <strong>
            Classi a cui sei iscritto
          </strong>
        </Text__BrandTxt>
      </Paragraph>
      <div>
        <Paragraph
          key="corso_1"
          theme={
            Object {
              "brand": "#00BBEF",
              "darken": "#1588AA",
            }
          }
        >
          corso 1
        </Paragraph>
        <Paragraph
          key="corso_2"
          theme={
            Object {
              "brand": "#00BBEF",
              "darken": "#1588AA",
            }
          }
        >
          corso 2
        </Paragraph>
      </div>
    </FormElements__CardForm>
  </Container>
</Page>
`;

exports[`<ClassiStudenteView /> visualizza il contenuto quando non ho corsi 1`] = `
<Page>
  <TopBar
    backArrow={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    backNav={
      Object {
        "enabled": true,
        "url": "/homepage",
      }
    }
    closeBtn={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    lesson={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    pinned={true}
    single={true}
    title="Classi"
    topSwitcher={
      Object {
        "items": Array [],
      }
    }
  />
  <Container>
    <FormElements__CardForm
      onSubmit={[Function]}
    >
      <Heading__H2>
        Associa classe
      </Heading__H2>
      <Paragraph
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        Inserisci il codice per associarti ad una classe
      </Paragraph>
      <FormElements__FormElement>
        <Field
          align="center"
          component={[Function]}
          label="codice"
          name="codice"
          type="text"
          validate={
            Array [
              [Function],
            ]
          }
        />
      </FormElements__FormElement>
      <FormElements__FormElement>
        <Button
          actioncolor="okay"
          full={true}
          theme={
            Object {
              "brand": "#00BBEF",
              "fonts": Object {
                "buttons": Object {
                  "fontFamily": "inherit",
                  "fontWeight": "inherit",
                },
              },
              "radius": Object {
                "buttons": "6px",
              },
            }
          }
        >
          Salva
        </Button>
      </FormElements__FormElement>
      <Paragraph
        margin="40px auto 20px"
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        <Text__BrandTxt>
          <strong>
            Classi a cui sei iscritto
          </strong>
        </Text__BrandTxt>
      </Paragraph>
      <AlertBanner
        actioncolor="help"
        theme={
          Object {
            "brand": "#00BBEF",
            "radius": Object {
              "general": "6px",
            },
          }
        }
      >
        Non sei iscritto a classi
      </AlertBanner>
    </FormElements__CardForm>
  </Container>
</Page>
`;

exports[`<ClassiStudenteView /> visualizza il contenuto quando spinner = true 1`] = `
<Page>
  <TopBar
    backArrow={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    backNav={
      Object {
        "enabled": true,
        "url": "/homepage",
      }
    }
    closeBtn={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    lesson={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    pinned={true}
    single={true}
    title="Classi"
    topSwitcher={
      Object {
        "items": Array [],
      }
    }
  />
  <Container>
    <Spinner
      number={3}
    />
  </Container>
</Page>
`;
